cg/meta/archive/archive.py

```diff
@@ -83,10 +99,111 @@ def archive_all_non_archived_spring_files(
+        files: List[File] = []
+        for sample in samples:
+            files.extend(
+                self.housekeeper_api.get_files_from_latest_version(
+                    bundle_name=sample.internal_id, tags=[SequencingFileTag.SPRING]
+                ).all()
+            )
+        return files
+
+    def add_housekeeper_destinations_to_samples(
+        self, samples: List[Sample]
+    ) -> List[SampleAndDestination]:
+        """Gets all samples which were run on the specified flowcell and clumps it with the destination in
+        Housekeeper where we want the retrieved files to be."""
+        samples_to_retrieve: List[SampleAndDestination] = []
+        for sample in samples:
+            LOG.info(f"Will try to retrieve sample: {sample.internal_id}.")
+            destination: str = self.get_destination_from_sample_internal_id(sample.internal_id)
+            samples_to_retrieve.append(SampleAndDestination(sample=sample, destination=destination))
+        return samples_to_retrieve
+
+    def retrieve_samples_from_archive_location(
+        self,
+        samples_and_destinations: List[SampleAndDestination],
+        archive_location: ArchiveLocations,
+    ):
+        archive_handler: ArchiveHandler = ARCHIVE_HANDLERS[archive_location](self.data_flow_config)
+        return archive_handler.retrieve_samples(samples_and_destinations)
+
+    def retrieve_sample(self, sample_internal_id: str) -> None:
+        """Retrieves all archived spring files for the given sample."""
+        sample: Sample = self.status_db.get_sample_by_internal_id(sample_internal_id)
+        archive_handler: ArchiveHandler = ARCHIVE_HANDLERS[sample.archive_location](
+            self.data_flow_config
+        )
+        housekeeper_destination: str = self.get_destination_from_sample_internal_id(
+            sample_internal_id
+        )
+        retrieval_task_id: int = archive_handler.retrieve_samples(
+            [SampleAndDestination(sample=sample, destination=housekeeper_destination)]
+        )
+        spring_files: List[File] = self.housekeeper_api.get_archived_files(
+            bundle_name=sample_internal_id, tags=[SequencingFileTag.SPRING]
+        )
+        self.set_archive_retrieval_task_ids(retrieval_task_id=retrieval_task_id, files=spring_files)
+
+    def get_destination_from_sample_internal_id(self, sample_internal_id) -> str:
+        """Returns where in Housekeeper to put the retrieved spring files for the specified sample."""
+        return self.housekeeper_api.get_latest_bundle_version(
+            sample_internal_id
+        ).full_path.as_posix()
+
+    def set_archive_retrieval_task_ids(self, retrieval_task_id: int, files: List[File]) -> None:
+        for file in files:
+            self.housekeeper_api.set_archive_retrieval_task_id(
+                file_id=file.id, retrieval_task_id=retrieval_task_id
+            )
+
+    def retrieve_spring_file(self, file_path: str):
```

Comment:
Same for this function. I think we should only add stuff we intend to use in the near future. But that's my fault for not being clear on how that is 😬